inmoose/limma/dups.py

```diff
@@ -0,0 +1,96 @@
+# -----------------------------------------------------------------------------
+# Copyright (C) 2004-2022 Gordon Smyth, Yifang Hu, Matthew Ritchie, Jeremy Silver, James Wettenhall, Davis McCarthy, Di Wu, Wei Shi, Belinda Phipson, Aaron Lun, Natalie Thorne, Alicia Oshlack, Carolyn de Graaf, Yunshun Chen, Mette Langaas, Egil Ferkingstad, Marcus Davy, Francois Pepin, Dongseok Choi
+# Copyright (C) 2024 Maximilien Colange
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+# This file is based on the file 'R/dups.R' of the Bioconductor limma package (version 3.55.1).
+
+import numpy as np
+
+
+def unwrapdups(M, ndups=2, spacing=1):
+    """
+    Unwraps a matrix (M) for a series of experiments, combining duplicate spots for each gene into a single row.
+
+    Arguments
+    ---------
+    M : ndarray
+        the input matrix containing data for multiple experiments.
+    ndups : int, optional
+        the number of duplicate spots for each gene. Defaults to 2.
+    spacing : int, optional
+        the number of rows between duplicate measurements for a gene. Defaults to 1.
+
+    Returns
+    -------
+    ndarray
+        the unwrapped matrix with duplicate spots combined.
+    """
+    if ndups == 1:
+        return M
```

Comment:
what happens if `ndups == 0`?  it return an empty ndarray?

Reply:
fixed